forest/benchmarking/direct_fidelity_estimation.py

```diff
@@ -171,223 +220,107 @@ def _monte_carlo_dfe(program: Program, qubits: Sequence[int], in_states: list, n
 
-    fid_var_est: float
-    """Variance of the fidelity point estimate, after considering the calibration."""
-
-    fid_std_err_est: float
-    """Standard deviation of the fidelity point estimate, after considering the calibration."""
+    The algorithm is due to:
 
+    [DFE1]  Practical Characterization of Quantum Devices without Tomography
+            Silva et al., PRL 107, 210404 (2011)
+            https://doi.org/10.1103/PhysRevLett.107.210404
 
-def direct_fidelity_estimate(data: DFEdata, cal: DFEdata, type: str) -> DFEestimate:
-    """
-    Estimate state or process fidelity by exhaustive direct fidelity estimation.
+    [DFE2]  Direct Fidelity Estimation from Few Pauli Measurements
+            Flammia and Liu, PRL 106, 230501 (2011)
+            https://doi.org/10.1103/PhysRevLett.106.230501
 
-    :param data: data from DFE experiment
-    :param cal: calibration data from DFE experiment
-    :param type: 'state' or 'process', 'process' is default.
+    :param program: A program comprised of Clifford group gates that constructs a state
+        for which we estimate the fidelity.
+    :param qubits: The qubits to perform DFE on. This can be a superset of the qubits
+        used in ``program``.
+    :param benchmarker: The `BenchmarkConnection` object used to design experiments
+    :param n_terms: Number of randomly chosen observables to measure. This number should be 
+        a constant less than ``2**len(qubits)``, otherwise ``exhaustive_process_dfe`` is more efficient.
     """
-    pauli_est = data.expectation / np.abs(cal.expectation)
-    var_est = ratio_variance(data.expectation, data.variance, cal.expectation, cal.variance)
-
-    temp_data = { 'dimension': data.dimension, 'expectation': pauli_est, 'variance': var_est }
-
-    if type == 'state':
-        fid = aggregate_state_dfe(temp_data)
-    elif type == 'process':
-        fid = aggregate_process_dfe(temp_data)
-    else:
-        raise ValueError('Error: must specify state or process DFE.')
-
-    return DFEestimate(
-        in_pauli=data.in_pauli,
-        program=data.program,
-        out_pauli=data.out_pauli,
-        dimension=data.dimension,
-        number_qubits=data.number_qubits,
-        pauli_point_est=pauli_est ,
-        pauli_var_est=var_est,
-        fid_point_est=fid['expectation'],
-        fid_var_est=fid['variance'],
-        fid_std_err_est=np.sqrt(fid['variance'])
-    )
-
-
-def aggregate_process_dfe(data: dict):
-    if isinstance(data['dimension'], int):
-        dim = data['dimension']
+    exp = TomographyExperiment(list(
+        _monte_carlo_dfe(program=program, qubits=qubits,
+                         in_states=[None, plusX, minusX, plusY, minusY, plusZ, minusZ],
+                         n_terms=n_terms, benchmarker=benchmarker)),
+        program=program, qubits=qubits)
+    return DFEexperiment(exp,'process')
+
+
+def acquire_dfe_data(qc: QuantumComputer, dfe: DFEexperiment, n_shots = 10_000, active_reset=False):
```

Comment:
Yeah I am not a fan of how the options in PyQuil are phrased.  I don't see a strong reason to default to `False`, actually. Should we just go for `True`? @karalekas @joshcombes  @kylegulshen

Reply:
Actually, I think the reason would be that active reset fidelity is not always as high as passive reset fidelity -- at least there is no guarantee it is as high.

Reply:
The reason given in the doc-string for `measure_observables` (in pyQuil) is that "Setting this to True is much faster but there is a ~1% error per qubit in the reset operation. Thermal noise from "traditional" reset is not routinely characterized but is of the same order."

Reply:
But since we're also on the topic of options re-phrasing options in pyQuil's `measure_observables`, I'm slightly peeved about the fact that the word `readout` appears at the beginning of the symmetrize option, and at the end for the calibrate option. So I'd vote for re-phrasing the option `readout_symmetrize` to `symmetrize_readout`, if we're making changes.

Reply:
Then let's move the conversation to pyQuil.